proxy/envoy/config_test.sh

```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+# Check that envoy can parse golden configuration artifacts
+# In the future, we can run functional tests, but all configs require iptables rules to be effective.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+ENVOY="proxy/envoy/envoy -l trace --service-cluster istio-proxy --service-node 10.0.0.1"
+CONFIG_FILES="envoy-v0 envoy-fault ingress-envoy egress-envoy"
+
+# TODO the test skips over SSL configuration files since referenced cert files are not in the right places
+
+for f in ${CONFIG_FILES}; do
```

Comment:
Thanks for checking re: namespaces. It looks only user namespaces are supported for now. The `exclusive` tag would solve the same problem and avoid running these tests with any other bazel test via serialization (see https://bazel.build/versions/master/docs/test-encyclopedia.html#tag-conventions).

sh_test supports `args` through the attributes common to all test rules (
https://bazel.build/versions/master/docs/be/common-definitions.html#common-attributes-tests).

With the above two changes the tests would look something like the following:

```bash
sh_test(
    name = "config_envoy_fault",
    size = "small",
    srcs = ["config_test.sh"],
    data = ["testdata/envoy-fault.json.golden"],
    args = ["testdata/envoy-fault.json.golden"],
    tags = ["exclusive"],
)
```

And then test output would look like this:

```bash
$ bazel test //proxy/envoy:config_envoy_v0 //proxy/envoy:config_envoy_fault
INFO: Found 2 test targets...
INFO: Elapsed time: 0.224s, Critical Path: 0.03s
//proxy/envoy:config_envoy_fault                                PASSED in 1.0s
//proxy/envoy:config_envoy_v0                                   PASSED in 1.0s

Executed 0 out of 2 tests: 2 tests pass.
```